### DISCUSSION TITLE
RSS Issues

### DISCUSSION THREAD

**Discussion by FuadEfendi** [hbstack/.github#73] [Issues and bugs]
I just noticed, RSS generated doesn't replace `{year}` with current year:
```
<rss xmlns:atom="http://www.w3.org/2005/Atom" version="2.0">
<channel>
<title>ABC</title>
<link>XYZ</link>
<description>ABC ...</description>
<generator>Hugo</generator>
<language>en</language>
<copyright>© 2004-{year} ... </copyright>
```

Also, would be nice to use default date (date of build) if `pubDate` is not defined, I am sure it is important for SEO:
```
<item>
<title>XYZ</title>
<link>https://....</link>
<pubDate>Mon, 01 Jan 0001 00:00:00 +0000</pubDate>
```

**Comment by razonyang** [comment]
> Also, would be nice to use default date (date of build) if pubDate is not defined, I am sure it is important for SEO:

~~Could you please provide some reference of `pubDate`, I remmember that it's property of items, not channel.~~

The `pubdate` already present, do you have customized the RSS template? For example, https://hbstack.dev/index.xml.

**Comment by razonyang** [comment]
RSS Copyright issue created hbstack/blog#946

**Comment by razonyang** [comment]
Copyright issue was fixed since [hbstack/blog@v0.37.2](https://github.com/hbstack/blog/releases/tag/v0.37.2).

I don't understand the `pubDate` request, please describe in details, if you mean the post doesn't have `date`, why wouldn't you set the `date` for it? It doesn't make sence, since the posts/items are ordered in weight/date, this will confuse users.

**Comment by FuadEfendi** [comment]
When front matter contains `date: 2024-08-17`, generated `index.xml` Atom RSS feed will contain
```
<item>
    ...
    <pubDate>Sat, 17 Aug 2024 00:00:00 +0000</pubDate>
    ...
</item>
```

**Reply by FuadEfendi** [reply]
and, when "date" is missing on front matter, RSS will have `<pubDate>Mon, 01 Jan 0001 00:00:00 +0000</pubDate>` - it may impact, for example, Google "Discover" feature for mobile devices (since date is in past)

So that I suggest to use default current date if date is not defined at front matter

**Reply by razonyang** [reply]
> It doesn't make sence, since the posts/items are ordered in weight/date, this will confuse users.

I see, but as I commented above, besides it's the duty of site owner, I'm not gonna implement it, you can either:

1. Set date manually/programmaticaly.
2. Use `cascade` to set default date for content.
3. CLI flag `--enableGitInfo` may also help.

As this, I'm going closing this discussion.

**Reply by FuadEfendi** [reply]
I see ;)

my pages typically have two dates... date defined in front matter, and another date from Git Commit. I didn't know about `cascade` options, thank you for your help.

**Reply by razonyang** [reply]
> I didn't know about cascade options

See https://gohugo.io/content-management/front-matter/#cascade, btw you can also define it in hugo.yaml.

You're welcome, thanks for reporting the bug.
